index.js

```diff
@@ -79,6 +82,125 @@ class BitbucketScm extends Scm {
+     *
+     * Searches through the webhook pages until the given webhook URL is found. If nothing is found, this will
+     * return nothing. If a status response of non-200 is encountered, the chain is rejected with the
+     * HTTP operation and the status code received.
+     * @method _findWebhook
+     * @param  {Object}     config
+     * @param  {Number}     config.page    pagination: page number to search next. 1-index.
+     * @param  {String}     config.repoId  The bitbucket repo ID (e.g., "username/repoSlug")
+     * @param  {String}     config.token   Admin Oauth2 token for the repo
+     * @param  {String}     config.url     url for webhook notifications
+     * @return {Promise}                   Resolves to a webhook information payload
+     */
+    _findWebhook(config) {
+        return this.breaker.runCommand({
+            url: `${API_URL_V2}/repositories/${config.repoId}/hooks?pagelen=30&page=${config.page}`,
+            method: 'GET',
+            login_type: 'oauth2',
+            oauth_access_token: config.token
+        }).then((response) => {
+            if (response.statusCode < 200 || response.statusCode >= 300) {
+                throw new Error(response.statusCode);
+            }
+
+            const hooks = response.body;
+            const result = hooks.values.find(webhook => webhook.url === config.url);
+
+            if (!result && hooks.size >= WEBHOOK_PAGE_SIZE) {
+                return this._findWebhook({
+                    page: config.page + 1,
+                    repoId: config.repoId,
+                    token: config.token,
+                    url: config.url
+                });
+            }
+
+            return result;
+        });
+    }
+
+    /**
+     * Creates and updates the webhook that is attached to a repo.
+     *
+     * By default, it creates a new webhook. If given a webhook payload, it will instead update the webhook to
+     * ensure the correct settings are in place. If a status response of non-200 is encountered, the chain is
+     * rejected with the HTTP operation and the status code received.
+     * @method _createWebhook
+     * @param  {Object}       config
+     * @param  {Object}       [config.hookInfo] Information about an existing webhook
+     * @param  {String}       config.repoId     Bitbucket repo ID (e.g., "username/repoSlug")
+     * @param  {String}       config.token      Admin Oauth2 token for the repo
+     * @param  {String}       config.url        url to create for webhook notifications
+     * @return {Promise}                        Resolves when complete
+     */
+    _createWebhook(config) {
+        const params = {
+            body: {
+                description: 'Screwdriver-CD build trigger',
+                url: config.url,
+                active: true,
+                events: [
+                    'repo:push',
+                    'pullrequest:updated',
+                    'pullrequest:created'
```

Comment:
"fullfilled" with two 'L's?

Looks like that's a bug: 

![screen shot 2016-12-07 at 2 57 01 pm](https://cloud.githubusercontent.com/assets/721148/20990408/71667e40-bc8d-11e6-802f-689ac55fde3c.png)

We'll update it to be the *correct* value in this PR.

Source: https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/hooks